src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -1141,25 +1141,27 @@ BOOL MethodTableBuilder::CheckIfSIMDAndUpdateSize()
     if (strcmp(className, "Vector`1") != 0 || strcmp(nameSpace, "System.Numerics") != 0)
         return false;
 
-    if (!TargetHasAVXSupport())
-        return false;
+    CORJIT_FLAGS CPUCompileFlags       = ExecutionManager::GetEEJitManager()->GetCPUCompileFlags();
+    uint32_t     numInstanceFieldBytes = 16;
+
+    if (!CPUCompileFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_PREJIT) && CPUCompileFlags.IsSet(InstructionSet_AVX2))
```

Comment:
It's unclear if the `!CPUCompileFlags.IsSet(CORJIT_FLAGS::CORJIT_FLAG_PREJIT)` can actually be hit. 

I only see it being set from `src\coreclr\tools\Common\JitInterface\CorInfoImpl.cs` but `GetMaxIntrinsicSIMDVectorLength` never looked to actually be called from R2R code.

I preserved it for the time being just in case as it was the safer option.

Reply:
It cannot be hit. It is left-over from NGen. It is 100% safe to delete it here.